src/fu-plugin.c

```diff
@@ -356,6 +356,20 @@ fu_plugin_device_add (FuPlugin *self, FuDevice *device)
 	g_return_if_fail (FU_IS_PLUGIN (self));
 	g_return_if_fail (FU_IS_DEVICE (device));
 
+	/* automatically set the device-id from the platform and logical ids */
+	if (fu_device_get_id (device) == NULL &&
+	    fu_device_get_physical_id (device) != NULL) {
+		g_autofree gchar *id = g_strjoin (":",
+						  fu_device_get_physical_id (device),
+						  fu_device_get_logical_id (device),
+						  NULL);
+		fu_device_set_id (device, id);
+	}
+	if (fu_device_get_id (device) == NULL) {
+		g_warning ("device was added with no ID, ignoring!");
```

Comment:
I think it would be good to add `fu_device_get_name` into this warning

Reply:
I don't think we can rely on name being set either, but we certainly can just dump the `_to_string()` output to screen for this pathological case.